design/fips-compliance.md

```diff
@@ -0,0 +1,259 @@
+ |----------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+ | Release only `goboring/golang` version | - Customers use the client and don't require FIPS compliancy, will now miss libraries that are available in `golang` but are not available in `goboring/golang`                                                                                                                                                                                                       |
+ | Release both versions                  | <ul><li>One more Dockerfile to maintain. Any change made to the original Dockerfile should be done also to `Dockerfile.fips` which can be easily forgotten</li><li>Another version to test. Our UTs will stay the same but we will need to run integration tests on both images, in order to release them. This adds complexity to our tests and doubles the tests run-time</li></ul> |
+
+From the table above, I think that we should release only the FIPS compliant
+version. The advantages that customers will get by using the `golang` image
+as the base image is not worth the maintenance burden that releasing both
+versions will bring to the development team.
+
+Note: If the performance of the `goboring/golang` version will be much worse than that
+      of the `golang` version (degradation of more than 5%) then we will release both versions.
+      More info on the performance tests that will be done in this effort can be found in the
+      [Performance section](#performance)
+
+### Red Hat image
+
+We will also release a FIPS compliant Red Hat image of the `conjur-authn-k8s
+-client` to our Red Hat registry. Before we do that, we should create a 
+`conjur-authn-k8s-client` FIPS project in the RH container registry.
+
+TBD - how do we create the FIPS project in the RH container registry?
+
+### Performance
+
+Same SLA of performance should be kept. As of today, we don't have any
+performance tests and an SLA (that I know of). We should decide on one and
+verify that the change to GoBoring doesn't impact the performance.
+
+We should write a performance test and run it twice - once on the `master` 
+image and once on the new FIPS-compliance image - and verify that the
+ performance is not affected by the change. We can live with a degradation of less
+ than 5% between the versions.
```

Comment:
From meeting: 
Hila + Dekel conversation - The 5% was in question and came to conclusion that this is subject to change and will be pushed to be pursued later on

Nahum - we should test  performance with FIPs and without FIPs